library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -353,6 +362,41 @@ public Slice get(Slice key, long timeBucket, ReadSource readSource)
       }
     }
 
+
+    private int filtedCount = 0;
+    private int unfiltedCount = 0;
+    private long verifyBloomFilterBeginTime = 0;
+    private long verifyBloomFilterInterval = 5000;
+    /**
+     * Test the result of using bloom filter and remove it if it not helpful for improving the performance.
+     *
+     * @param mightContain true if collection might contain the value
+     */
+    private void verifyBloomFilter(boolean mightContain)
+    {
+      if (!mightContain) {
+        filtedCount++;
+      } else {
+        unfiltedCount++;
+      }
+      long now = System.currentTimeMillis();
```

Comment:
@davidyan74 this change only impact performance but don't impact functional. So I think it does not break the idempotency.